### ISSUE TITLE
If a command has no args, only opts, it is not possible to use `--help`

### ISSUE THREAD

**Issue by fabjan** [fabjan/outil#11]
I'm guessing this has something to do with the help text feature (due to lazy design) piggybacking on error handling.

Options are always optional so maybe I missed something about them when I implemented `--help`.

Fix the bug and add a test for `Help(_)` given a command with no args, just option(s).

**Comment by fabjan** [comment]
Looking at the comment in the implementation:

```
  // This solution (piggybacking on the error handling) is not ideal since it
  // means we can only show help for commands that try to parse any arguments
  // or options. But if your command doesn't take any arguments or options
  // then you probably don't need this library at all.
```

Well that was incorrect! Options are optional so they might never be passed or parsed.

**Comment by fabjan** [comment]
Fixed by #12
